bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -245,20 +245,25 @@ public void run() {
     int timeout = retryOptions.getReadPartialRowTimeoutMillis();
 
     int streamingBufferSize = retryOptions.getStreamingBufferSize();
-    int batchRequestSize = streamingBufferSize / 4;
+    int batchRequestSize = retryOptions.getStreamingBatchSize();
     ResponseQueueReader responseQueueReader =
         new ResponseQueueReader(timeout, streamingBufferSize, batchRequestSize,
             batchRequestSize, readRowsCall);
 
+    System.out.println(String.format("Buffer size: %d, Batch size: %d", streamingBufferSize,
+      batchRequestSize));
     StreamingBigtableResultScanner resultScanner =
         new StreamingBigtableResultScanner(reservedChannel, responseQueueReader, cancellationToken);
 
     clientCallService.asyncServerStreamingCall(
         readRowsCall,
         request,
-        batchRequestSize,
         createClientCallListener(resultScanner));
 
+    if (batchRequestSize > 1) {
```

Comment:
Yes.  It made more sense to have the code here rather thanClientCallService.java.